sdk/lib/rtl/heap.c

```diff
@@ -1163,7 +1181,8 @@ RtlpCoalesceFreeBlocks (PHEAP Heap,
         !(CurrentEntry->Flags & HEAP_ENTRY_BUSY) &&
         (*FreeSize + CurrentEntry->Size) <= HEAP_MAX_BLOCK_SIZE)
     {
-        ASSERT(FreeEntry->PreviousSize == CurrentEntry->Size);
+        if (CurrentEntry->Size)
```

Comment:
Can you explain this check? What does a size of 0 mean? Is that even possible?

Reply:
This was asserting with CurrentEntry->Size == 0 before the handling for segments > MAXUSHORT was fixed. It may not be needed now. I added a comment stating this. Thanks.